includes/config.inc

```diff
@@ -540,6 +542,76 @@ function islandora_solr_metadata_config_field_form($form, &$form_state, $config_
 }
 
 /**
+ * Genereate the element definition for the permission fieldset.
+ *
+ * @param bool|array $permissions
+ *   The selected permissions (from the $form_state) either an array containing
+ *   key values, or FALSE if it is not enabled.
+ * @param array $permissions_default
+ *   The permissions to select by default.
+ * @param array $permissions_disable
+ *   Some permissions which we will not allow to be changed (set as disabled in
+ *   the form).
+ * @param bool $default_value
+ *   Whether the checkbox is to be checked by default.
+ *
+ * @return array
+ *   An associative array containing the definition for the permissions
+ *   fieldset.
+ */
+function islandora_solr_metadata_get_admin_permissions_fieldset($permissions, $permissions_default, $permissions_disable, $default_value) {
+  $default_enable = array_sum($permissions) > 0 ? TRUE : $default_value;
```

Comment:
Not sure if this is appropriate... Would end up disabling permission enforcement if all roles were denied access to a field... Not where _when_ this might happen, but there may be some kind of crazy use cases, or for testing things?